gstorm/include/detail/ranges/vector.h

```diff
@@ -45,182 +40,187 @@ struct range_forward_traits {
+  }
 
-    friend iterator operator+(const iterator &lhs, difference_type n) {
-      if (!lhs.hasAccessor())
-        return iterator(lhs.it + n);
-      return iterator(lhs.accessor.value(), lhs.it + n, lhs.owner);
-    }
+  friend iterator operator+(const iterator &lhs, difference_type n) {
+    return iterator(lhs.accessor, lhs.it + n, lhs.owner);
+  }
 
-    friend iterator operator+(difference_type n, const iterator &rhs) {
-      if (!rhs.hasAccessor())
-        return iterator(rhs.it + n);
-      return iterator(rhs.accessor.value(), rhs.it + n, rhs.owner);
-    }
+  friend iterator operator+(difference_type n, const iterator &rhs) {
+    return iterator(rhs.accessor, rhs.it + n, rhs.owner);
+  }
 
-    friend iterator operator-(const iterator &lhs, difference_type n) {
-      if (!lhs.hasAccessor())
-        return iterator(lhs.it - n);
-      return iterator(lhs.accessor.value(), lhs.it - n, lhs.owner);
-    }
+  friend iterator operator-(const iterator &lhs, difference_type n) {
+    return iterator(lhs.accessor, lhs.it - n, lhs.owner);
+  }
 
-    friend difference_type operator-(const iterator &left, const iterator &right) {
-      return left.it - right.it;
-    }
+  friend difference_type operator-(const iterator &left, const iterator &right) {
+    return left.it - right.it;
+  }
 
-    friend iterator &operator-=(iterator &lhs, difference_type n) {
-      lhs.it -= n;
-      return lhs;
-    }
+  friend iterator &operator-=(iterator &lhs, difference_type n) {
+    lhs.it -= n;
+    return lhs;
+  }
 
-    friend iterator &operator+=(iterator &lhs, difference_type n) {
-      lhs.it += n;
-      return lhs;
-    }
+  friend iterator &operator+=(iterator &lhs, difference_type n) {
+    lhs.it += n;
+    return lhs;
+  }
 
-    friend bool operator<(const iterator &left, const iterator &right) {
-      return left.it < right.it;
-    }
+  friend bool operator<(const iterator &left, const iterator &right) {
+    return left.it < right.it;
+  }
 
-    friend bool operator>(const iterator &left, const iterator &right) {
-      return left.it > right.it;
-    }
+  friend bool operator>(const iterator &left, const iterator &right) {
+    return left.it > right.it;
+  }
 
-    friend bool operator<=(const iterator &left, const iterator &right) {
-      return left.it <= right.it;
-    }
+  friend bool operator<=(const iterator &left, const iterator &right) {
+    return left.it <= right.it;
+  }
 
-    friend bool operator>=(const iterator &left, const iterator &right) {
-      return left.it >= right.it;
-    }
+  friend bool operator>=(const iterator &left, const iterator &right) {
+    return left.it >= right.it;
+  }
 
-    bool operator==(const iterator &other) const { return other.it == it; }
+  bool operator==(const iterator &other) const { return other.it == it; }
 
-    bool operator!=(const iterator &other) const { return other.it != it; }
+  bool operator!=(const iterator &other) const { return other.it != it; }
```

Comment:
Please implement

* `operator!=` in terms of `operator==`
* `operator<=`, `operator>` and `operator>=` in terms of `operator<`